src/Features/CSharp/Portable/RemoveUnusedParametersAndValues/CSharpRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -0,0 +1,235 @@
+                    localDeclarationStatement.Declaration.Variables.Any(IsDiscardDeclaration))
+                {
+                    RemoveDiscardHelper.ProcessDeclarationStatement(localDeclarationStatement, editor);
+                }
+                else if (child is CatchDeclarationSyntax catchDeclaration &&
+                    IsDiscardDeclaration(catchDeclaration))
+                {
+                    // "catch (Exception _)" => "catch (Exception)"
+                    editor.ReplaceNode(catchDeclaration, catchDeclaration.WithIdentifier(default));
+                }
+            }
+
+            return Task.FromResult(editor.GetChangedRoot());
+        }
+
+        private static bool IsDiscardDeclaration(VariableDeclaratorSyntax variable)
+            => variable.Identifier.Text == AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.DiscardVariableName;
+        private static bool IsDiscardDeclaration(CatchDeclarationSyntax catchDeclaration)
+            => catchDeclaration.Identifier.Text == AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.DiscardVariableName;
+
+        private sealed class RemoveDiscardHelper : IDisposable
+        {
+            private readonly LocalDeclarationStatementSyntax _localDeclarationStatement;
+            private readonly SyntaxEditor _editor;
+            private readonly ArrayBuilder<StatementSyntax> _statementsBuilder;
+            private SeparatedSyntaxList<VariableDeclaratorSyntax> _currentNonDiscardVariables;
+
+            private RemoveDiscardHelper(LocalDeclarationStatementSyntax localDeclarationStatement, SyntaxEditor editor)
+            {
+                _localDeclarationStatement = localDeclarationStatement;
+                _editor = editor;
+
+                _statementsBuilder = ArrayBuilder<StatementSyntax>.GetInstance();
+                _currentNonDiscardVariables = new SeparatedSyntaxList<VariableDeclaratorSyntax>();
+            }
+
+            public static void ProcessDeclarationStatement(
+                LocalDeclarationStatementSyntax localDeclarationStatement,
+                SyntaxEditor editor)
+            {
+                using (var helper = new RemoveDiscardHelper(localDeclarationStatement, editor))
+                {
+                    helper.ProcessDeclarationStatement();
+                }
+            }
+
+            public void Dispose() => _statementsBuilder.Free();
+
+            private void ProcessDeclarationStatement()
+            {
+                // We will replace all discard variable declarations in this method with discard assignments,
+                // For example,
+                //  1. "var _ = M();" is replaced with "_ = M();"
+                //  2. "var x = 1, _ = M(), y = 2;" is replaced with following statements:
+                //          var x = 1;
+                //          _ = M();
+                //          var y = 2;
```

Comment:
'1' seems fine.  '2' seems onerous.   Note: it's not clear to me how you even have '2' since you have to have a type name when being multi-initialized.  Though maybe you meant this is what you'd get with `int x ...`?

Reply:
Yes, fixed the example to have explicit type, not var. '2' is indeed very unlikely, but I still implemented it given that it is allowed by the language and will lead to compiler errors (not just warning) if we ignored it.

---
In reply to: [234799453](https://github.com/dotnet/roslyn/pull/30777#discussion_r234799453) [](ancestors = 234799453)